ffi/src/scan.rs

```diff
@@ -112,27 +123,28 @@ unsafe fn scan_impl(
             scan_builder = scan_builder.with_predicate(predicate);
         }
     }
-    Ok(BoxHandle::into_handle(scan_builder.build()))
+    Ok(Box::new(scan_builder.build()).into())
 }
 
-impl BoxHandle for GlobalScanState {}
+#[handle_descriptor(target=GlobalScanState, mutable=true, sized=true)]
+pub struct GlobalScanStateHandle;
```

Comment:
Rescuing https://github.com/delta-incubator/delta-kernel-rs/pull/182#discussion_r1603863446
> Not sure what prefix we want to use instead of "Shared" for `Mutable = True` things. Maybe `Mut`, or maybe `Exclusive`?

Shouldn't we just have the rule that it's mutable unless clearly stated otherwise? That kind of "opt out" is safer than an "opt in" approach?

Reply:
Yeah, for this one it's probably fine because we can use `GlobalScanStateHandle*` and don't have to use `HandleGlobalScanStateHandle`.

For some of the `ExternResult` stuff the types get... not so nice (i.e. `ExternResultHandleKernelScanDataIteratorHandle`)

Reply:
As in, if it doesn't say "shared" in the name... it's not shared.